src/utils/tableToImage.ts

```diff
@@ -0,0 +1,238 @@
+ * @param markdown - The markdown table content
+ * @return HTML string
+ */
+async function renderMarkdownToHtml(markdown: string): Promise<string> {
+	// Parse markdown table to HTML
+	const lines = markdown.trim().split('\n')
+	if (lines.length < 2) {
+		throw new Error('Invalid table format')
+	}
+
+	let html = `<table style="${TABLE_STYLE}">`
+
+	// Parse header
+	const headerCells = lines[0].split('|').filter(cell => cell.trim())
+	html += '<thead><tr>'
+	headerCells.forEach(cell => {
+		html += `<th style="${HEADER_CELL_STYLE}">${parseInlineMarkdown(cell.trim())}</th>`
+	})
+	html += '</tr></thead>'
+
+	// Skip separator line (index 1)
+	// Parse body rows
+	html += '<tbody>'
+	for (let i = 2; i < lines.length; i++) {
+		const cells = lines[i].split('|').filter(cell => cell.trim())
+		if (cells.length > 0) {
+			html += '<tr>'
+			cells.forEach(cell => {
+				html += `<td style="${CELL_BASE_STYLE}">${parseInlineMarkdown(cell.trim())}</td>`
+			})
+			html += '</tr>'
+		}
+	}
+	html += '</tbody></table>'
+
+	return html
+}
+
+/**
+ * Parse inline markdown formatting (bold, italic, code, strikethrough, etc.)
+ * @param text - The text to parse
+ * @return HTML string with inline formatting
+ */
+function parseInlineMarkdown(text: string): string {
+	let result = text
+
+	// Escape HTML special characters first (except & which might be part of existing entities)
+	result = result
+		.replace(/</g, '&lt;')
+		.replace(/>/g, '&gt;')
+
+	// Bold with ** or __
+	result = result.replace(/\*\*(.+?)\*\*/g, '<strong>$1</strong>')
+	result = result.replace(/__(.+?)__/g, '<strong>$1</strong>')
+
+	// Italic with * or _
+	result = result.replace(/\*(.+?)\*/g, '<em>$1</em>')
+	result = result.replace(/_(.+?)_/g, '<em>$1</em>')
+
+	// Strikethrough with ~~
+	result = result.replace(/~~(.+?)~~/g, '<del>$1</del>')
+
+	// Inline code with `
+	result = result.replace(/`(.+?)`/g, `<code style="${CODE_STYLE}">$1</code>`)
+
+	// Links [text](url)
+	result = result.replace(/\[([^\]]+)\]\(([^)]+)\)/g, `<a href="$2" style="${LINK_STYLE}">$1</a>`)
+
+	return result
+}
+
+/**
+ * Convert HTML to a data URL using canvas rendering
+ * @param html - The HTML content to convert
+ * @return Data URL of the rendered image
+ */
+async function htmlToDataUrl(html: string): Promise<string> {
+	return new Promise((resolve, reject) => {
+		// Create a temporary container
+		const container = document.createElement('div')
+		container.innerHTML = html
+		container.style.position = 'absolute'
+		container.style.left = '-9999px'
+		container.style.top = '-9999px'
+		container.style.padding = '16px'
+		container.style.backgroundColor = 'white'
+		document.body.appendChild(container)
+
+		// Wait for next frame to ensure rendering
+		requestAnimationFrame(() => {
+			const cleanup = () => document.body.removeChild(container)
+			try {
+				// Use html2canvas if available, otherwise use a simple SVG approach
+				if (typeof window.html2canvas === 'function') {
```

Comment:
Is this something that some browsers offer? Otherwise a reasonable dependency to add?